### PR TITLE
QA-333: enable email-dependant integration tests running in staging

### PR DESCRIPTION
**Pull request by tranchitella**
Enable email-dependant integration tests when running in staging for
both backend and full integration tests. It depends on the GMAIL_ADDRESS
and GMAIL_PASSWORD env variables, which point to a dedicated gmail
mailbox which will receive the emails from the staging environment.

For this reason, the users created by these tests are in the from:

  ci.email.tests+UUIDv4@mender.io

Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

Example pipeline:
* https://gitlab.com/Northern.tech/Mender/integration/-/pipelines/449095015

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Review by kjaskiewiczz — APPROVED** [review]

**Review by oleorhagen — APPROVED** [review]

File: testutils/api/client.py [hunk 1]
```diff
@@ -27,6 +28,11 @@
 GATEWAY_HOSTNAME = os.environ.get("GATEWAY_HOSTNAME") or "mender-api-gateway"
 
 
+def get_free_tcp_port() -> int:
```
Comment (MaciejTe) [inline testutils/api/client.py:31]: Shouldn't this be located somewhere in `testutils` (for example, `testutils/util/network.py`) so backend tests also can access it?

File: backend-tests/tests/conftest.py [hunk 2]
```diff
@@ -23,14 +23,17 @@
...
+from testutils.api.client import wait_for_port
 
 
 urllib3.disable_warnings()
 
 wait_until_healthy("backend-tests")
 
 
-forward_port = 8080
+def get_free_tcp_port() -> int:
```
Comment (MaciejTe) [inline backend-tests/tests/conftest.py:34]: Shouldn't this be located somewhere in testutils (for example, testutils/util/network.py) so we don't replicate code?

File: testutils/infra/smtpd_mock.py [hunk 3]
```diff
@@ -76,8 +93,80 @@ def assert_called(self, email):
...
+                return
+            for i in range(int(mail_ids[-1]), int(mail_ids[0]) - 1, -1):
+                mail.store(str(i), "+FLAGS", "\\Deleted")
+            mail.expunge()
+        finally:
+            mail.close()
+
+
 @pytest.fixture(scope="function")
 def smtp_mock():
```
Comment (MaciejTe) [inline testutils/infra/smtpd_mock.py:163]: Technically it's not mock, I suggest to change fixture name

File: testutils/infra/smtpd_mock.py [hunk 4]
```diff
@@ -76,8 +93,80 @@ def assert_called(self, email):
...
+    global GMAIL_SERVER, GMAIL_ADDRESS, GMAIL_PASSWORD
```
Comment (MaciejTe) [inline testutils/infra/smtpd_mock.py:164]: Nitpick: using global variables is considered a bad practice - is this really necessary?

**Comment by lluiscampos** [comment]
Enough eyes on this one, I leave the review to the others :+1:

File: testutils/infra/smtpd_mock.py (hunk 3 above)
Comment (tranchitella) [reply testutils/infra/smtpd_mock.py:163]: This was the original name of the fixture (which was a mock).
I'm renaming it smtp_server.

**Review by MaciejTe — APPROVED** [review]

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Pull request merged**